VMEncryption/main/DiskUtil.py

```diff
@@ -1091,7 +1087,24 @@ def is_os_disk_lvm(self):
 
         return DiskUtil.os_disk_lvm
 
-    def should_skip_for_inplace_encryption(self, device_item, encrypt_volume_type):
+    def is_data_disk(self, device_item, azure_devices):
+        # Root disk
+        if device_item.device_id.startswith('00000000-0000'):
+            self.logger.log(msg="skipping root disk", level=CommonVariables.WarningLevel)
+            return False
+        # Resource Disk. Not considered a "data disk" exactly (is not attached via portal and we have a separate code path for encrypting it)
+        if device_item.device_id.startswith('00000000-0001'):
+            self.logger.log(msg="skipping resource disk", level=CommonVariables.WarningLevel)
+            return False
+
+        for azure_blk_item in azure_devices:
+            if azure_blk_item.name == device_item.name:
+                self.logger.log(msg="the mountpoint is the azure disk root or resource, so skip it.")
```

Comment:
If the device was root or resource, wouldn't we have already logged that we skipped it in the prior two if conditions?   The devices listed in azure_devices are not known to this function, maybe the log message can inject the device item name it is skipping into the message.  Maybe this skips the BEK volume?

Reply:
Commented before reading your prior remark about Gen 1 and Gen 2.   So maybe add that as a comment in the code that the first checks are for Gen1 and the second checks are for Gen2 images.   Also ... Is there any other type of special device here other than root or resource that might be included in this list?  It is passed dynamically to the function. Would like to make sure BEK volume is not going to be misidentified as a data disk.